app/trace.py

```diff
@@ -0,0 +1,66 @@
+TRACEBACK_ID = 'Traceback (most recent call last):'
+PARSE_ERROR_ID = 'CoconutParseError: parsing failed'
+
+def extract_trace_py(output):
+    """Extracts info from python trace output
+    Returns line numbers of errors.
+    """
+    capture = False
+    tracebacks = []
+    curr = {}
+
+    lines = output.split('\n')
+    for index, line in enumerate(lines):
+        if TRACEBACK_ID in line:
+            capture = True
+        elif capture and line.startswith(' '):
+            if line.lstrip().startswith('File'):
+                # Line with file, line number, and context
+                line = line.strip()
+                fname, line_num, context = [s.strip() for s in line.split(',')]
+                # Next line is the called function
+                call = lines[index+1].strip()
+                # Update current trace
+                curr['file'] = fname.lstrip('File ')[1:-1]
+                curr['line'] = int(line_num.lstrip('line '))
+                curr['context'] = context.lstrip('in ')
+                curr['call'] = call
+                # Offset line number by length of coconut header
+                curr['line'] -= 637
```

Comment:
Can you make sure that the length of coconut header is the same for all code input.

Reply:
Actually, can you think of the other ways we don't have to offset with specific line numbers? I don't think it would work when the coconut header is changed in the future.

Reply:
From http://coconut.readthedocs.io/en/master/DOCS.html#compilation-modes:
> if only a single file is compiled, that information will be stored within a header inside the file (standalone mode).

This is the header length for standalone files, which is the default since we run coconut on the written file.

Reply:
Can you count how many lines based on this code snippet in `app.py`

```
SEPARATOR = "# Compiled Coconut: -----------------------------------------------------------\n\n"
python_code = compiled_code.split(SEPARATOR)[-1]
```

Reply:
Yeah, we can instead only support coconut parse errors for now.

Reply:
Something like:
```python
SEPARATOR = "# Compiled Coconut: -----------------------------------------------------------\n\n"
split_code = compiled_code.split(SEPARATOR)
header, python_code = split_code[:-1], split_code[-1]
len(header)
```

Reply:
Ok. Can you also try the code above and submit a PR in the future?

Reply:
Done